### ISSUE TITLE
Exception when using aggregations and FLS

### ISSUE THREAD

**Issue by JulienCarnec** [opendistro-for-elasticsearch/security-advanced-modules#14]
We get exceptions when using Field Level Security and aggregations.

Here is the use-case:
- I have 2 users
    - **admin**, with full access
    - **restricted**, which anonymise a field named 'state' in the documents:
        - in the Kibana UI, 
            - The cluster role permission is set to CLUSTER_COMPOSITE_OPS_RO
            - The index permission action group is set to READ
            - The only change in the DLS/FLS tab is to declare 'state' as the field to anonymize

![image](https://user-images.githubusercontent.com/18028134/57783377-4fa62400-772e-11e9-985e-aede6cbf3850.png)

When I run the following query
```
{  
  "stored_fields":"*",
  "docvalue_fields":[
    {  
      "field":"timestamp",
      "format":"date_time"
    }
  ],
  "aggregations":{  
    "2":{  
      "terms":{  
        "field":"processName.keyword"
      }
    }
  }
}
```

It runs fine with **admin**, but **restricted** user gets the following error:
```
{
  "took": 59,
  "timed_out": false,
  "_shards": {
    "total": 10,
    "successful": 1,
    "skipped": 0,
    "failed": 9,
    "failures": [{
      "shard": 0,
      "index": "processes-active-idx-2019.04.25-000001",
      "node": "khHAhpFEQT2hyCgClyfojQ",
      "reason": {
        "type": "no_such_element_exception",
        "reason": "no_such_element_exception: No value present"
      }
    }, {
      "shard": 0,
      "index": "processes-completed-idx-2019.04.25-000001",
      "node": "khHAhpFEQT2hyCgClyfojQ",
      "reason": {
        "type": "no_such_element_exception",
        "reason": "no_such_element_exception: No value present"
      }
    }]
  },
  "hits": {
    "total": 0,
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "2": {
      "doc_count_error_upper_bound": 0,
      "sum_other_doc_count": 0,
      "buckets": []
    }
  }
}
```

Which corresponds to the following exception in Elasticsearch:
```
[2019-05-15T13:25:43,107][DEBUG][o.e.a.s.TransportSearchAction] [elastic-client-5d87cccdcb-kcpvz] [processes-completed-idx-2019.04.25-000001][4], node[khHAhpFEQT2hyCgClyfojQ], [P], s[STARTED], a[id=2LVtroVWRqaxnw29IQB5gg]: Failed to execute [SearchRequest{searchType=QUERY_THEN_FETCH, indices=[processes-*], indicesOptions=IndicesOptions[ignore_unavailable=false, allow_no_indices=true, expand_wildcards_open=true, expand_wildcards_closed=false, allow_aliases_to_multiple_indices=true, forbid_closed_indices=true, ignore_aliases=false, ignore_throttled=true], types=[], routing='null', preference='null', requestCache=false, scroll=null, maxConcurrentShardRequests=5, batchedReduceSize=512, preFilterShardSize=128, allowPartialSearchResults=true, source={"stored_fields":"*","docvalue_fields":[{"field":"timestamp","format":"date_time"}],"aggregations":{"2":{"terms":{"field":"processName.keyword","size":10,"min_doc_count":1,"shard_min_doc_count":0,"show_term_doc_count_error":false,"order":[{"_count":"desc"},{"_key":"asc"}]}}}}}]
org.elasticsearch.transport.RemoteTransportException: [elastic-data-0][10.1.29.246:9300][indices:data/read/search[phase/query]]
Caused by: org.elasticsearch.common.io.stream.NotSerializableExceptionWrapper: no_such_element_exception: No value present
  at java.util.Optional.get(Optional.java:135) ~[?:1.8.0_212]
  at com.amazon.opendistroforelasticsearch.security.configuration.DlsFlsFilterLeafReader.wrapSortedSetDocValues(DlsFlsFilterLeafReader.java:821) ~[?:?]
  at com.amazon.opendistroforelasticsearch.security.configuration.DlsFlsFilterLeafReader.getSortedSetDocValues(DlsFlsFilterLeafReader.java:810) ~[?:?]
  at org.apache.lucene.index.DocValues.getSortedSet(DocValues.java:429) ~[lucene-core-7.6.0.jar:7.6.0 719cde97f84640faa1e3525690d262946571245f - nknize - 2018-12-07 14:44:20]
  at org.elasticsearch.index.fielddata.plain.SortedSetDVBytesAtomicFieldData.getOrdinalsValues(SortedSetDVBytesAtomicFieldData.java:52) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.index.fielddata.ordinals.GlobalOrdinalsIndexFieldData$Atomic.getOrdinalsValues(GlobalOrdinalsIndexFieldData.java:135) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.aggregations.support.ValuesSource$Bytes$WithOrdinals$FieldData.globalOrdinalsValues(ValuesSource.java:153) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.aggregations.support.ValuesSource$Bytes$WithOrdinals.globalMaxOrd(ValuesSource.java:124) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorFactory.getMaxOrd(TermsAggregatorFactory.java:215) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorFactory.doCreateInternal(TermsAggregatorFactory.java:137) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.aggregations.support.ValuesSourceAggregatorFactory.createInternal(ValuesSourceAggregatorFactory.java:59) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.aggregations.AggregatorFactory.create(AggregatorFactory.java:216) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.aggregations.AggregatorFactories.createTopLevelAggregators(AggregatorFactories.java:218) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.aggregations.AggregationPhase.preProcess(AggregationPhase.java:55) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:112) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:351) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:398) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.SearchService.access$100(SearchService.java:126) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:360) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:356) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.search.SearchService$4.doRun(SearchService.java:1117) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:759) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:41) ~[elasticsearch-6.6.2.jar:6.6.2]
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.6.2.jar:6.6.2]
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_212]
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_212]
  at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

The exceptions are raised from this code: 
https://github.com/opendistro-for-elasticsearch/security-advanced-modules/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/DlsFlsFilterLeafReader.java

If I remove the `aggregations` from the query, it runs fine with both users.
If I aggregate on the anonymized field `state.keyword` instead of `processName.keyword`, it also runs fine with both users.

Versions used:
- Elasticsearch 6.6.2
- OpenDistro security plugin 0.8.0.0

**Comment by JulienCarnec** [comment]
I noticed that when I save a Role in Kibana, a toaster appears stating that the role was successfully saved...

...but a few seconds later, the Elasticsearch pod which received the query from Kibana issues the following exception:
```
[2019-05-16T19:43:46,413][WARN ][r.suppressed             ] [bai-test-ibm-dba-ek-client-575649c7-6w6dw] path: /_cluster/health, params: {}
org.elasticsearch.transport.RemoteTransportException: [bai-test-ibm-dba-ek-master-0][10.1.29.232:9300][cluster:monitor/health]
Caused by: org.elasticsearch.ElasticsearchSecurityException: Unexpected exception cluster:monitor/health
	at com.amazon.opendistroforelasticsearch.security.filter.OpenDistroSecurityFilter.apply0(OpenDistroSecurityFilter.java:274) ~[?:?]
	at com.amazon.opendistroforelasticsearch.security.filter.OpenDistroSecurityFilter.apply(OpenDistroSecurityFilter.java:119) ~[?:?]
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:165) ~[elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:139) ~[elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:83) ~[elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:73) ~[elasticsearch-6.6.2.jar:6.6.2]
	at com.amazon.opendistroforelasticsearch.security.ssl.transport.OpenDistroSecuritySSLRequestHandler.messageReceivedDecorate(OpenDistroSecuritySSLRequestHandler.java:194) ~[?:?]
	at com.amazon.opendistroforelasticsearch.security.transport.OpenDistroSecurityRequestHandler.messageReceivedDecorate(OpenDistroSecurityRequestHandler.java:284) ~[?:?]
	at com.amazon.opendistroforelasticsearch.security.ssl.transport.OpenDistroSecuritySSLRequestHandler.messageReceived(OpenDistroSecuritySSLRequestHandler.java:166) ~[?:?]
	at com.amazon.opendistroforelasticsearch.security.OpenDistroSecurityPlugin$7$1.messageReceived(OpenDistroSecurityPlugin.java:652) ~[?:?]
	at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:66) ~[elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.transport.TcpTransport$RequestHandler.doRun(TcpTransport.java:1289) ~[elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.util.concurrent.EsExecutors$1.execute(EsExecutors.java:140) ~[elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.transport.TcpTransport.handleRequest(TcpTransport.java:1247) [elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.transport.TcpTransport.messageReceived(TcpTransport.java:1111) [elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:914) [elasticsearch-6.6.2.jar:6.6.2]
	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:53) [transport-netty4-client-6.6.2.jar:6.6.2]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:241) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1436) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1203) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1247) [netty-handler-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278) [netty-codec-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:656) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:556) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:510) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:470) [netty-transport-4.1.32.Final.jar:4.1.32.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:909) [netty-common-4.1.32.Final.jar:4.1.32.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

**Comment by JulienCarnec** [comment]
Fixed by upgrading to OpenDistro 0.9.0
